src/rollup/plugins/externals.ts

```diff
@@ -197,102 +188,187 @@ export function externals(opts: NodeExternalsOptions): Plugin {
+          await fsp.copyFile(src, dst);
         }
-      }
 
-      const writeFile = async (file: string) => {
-        if (!(await isFile(file))) {
-          return;
-        }
-        const src = resolve(opts.traceOptions.base, file);
-        const { pkgName, subpath } = parseNodeModulePath(file);
-        const dst = resolve(opts.outDir, `node_modules/${pkgName + subpath}`);
+        // Copy package.json
+        const pkgJSONPath = join(
+          opts.outDir,
+          "node_modules",
+          outputName || pkg.name,
+          "package.json"
+        );
+        await fsp.mkdir(dirname(pkgJSONPath), { recursive: true });
+        await fsp.copyFile(
+          join(pkg.versions[version].path, "package.json"),
+          pkgJSONPath
+        );
+      };
+
+      const linkPackage = async (from: string, to: string) => {
+        const src = join(opts.outDir, "node_modules", from);
+        const dst = join(opts.outDir, "node_modules", to);
         await fsp.mkdir(dirname(dst), { recursive: true });
-        try {
-          await fsp.copyFile(src, dst);
-        } catch {
-          consola.warn(`Could not resolve \`${src}\`. Skipping.`);
-        }
+        // TODO: Windows workaround with junctions?
+        await fsp.symlink(src, dst);
       };
 
-      // Write traced files
+      // Write traced packages
       await Promise.all(
-        tracedFiles.map((file) => retry(() => writeFile(file), 3))
+        Object.values(tracedPackages).map(async (tracedPackage) => {
+          const versions = Object.keys(tracedPackage.versions); // TODO: sort by semver
+          if (versions.length === 1) {
+            // Write the only version into node_modules/{name}
+            await writePackage(tracedPackage.name, versions[0]);
+          } else {
+            for (const version of versions) {
+              // Try to find parent packages
+              const versionFiles: TracedFile[] = tracedPackage.versions[
+                version
+              ].files.map((path) => tracedFiles[path]);
+              const parentPkgs = [
+                ...new Set(
+                  versionFiles.flatMap((file) =>
+                    file.parents.flatMap(
+                      (parentPath) => tracedFiles[parentPath].pkgName
+                    )
+                  )
+                ),
+              ];
+
+              if (parentPkgs.length === 0) {
+                // No parent packages, assume as the hoisted version
+                await writePackage(tracedPackage.name, version);
+              } else {
+                // Write alternative version into node_modules/{name}@{version}
+                await writePackage(
+                  tracedPackage.name,
+                  version,
+                  `${tracedPackage.name}@${version}`
+                );
+                // For each parent, link into node_modules/{parent}/node_modules/{name}
+                for (const parentPath of parentPkgs) {
```

Comment:
Linking breaks if the nested packages also have conflicting children.
Consider the following structure:

```
└── node_modules
    ├── nitro-dep-a
    │   └── nitro-lib#v1.0.0
    │       └── nested-lib#v1.0.0
    ├── nitro-dep-b
    │   └── nitro-lib#v2.0.0
    │       └── nested-lib#v2.0.0
    └── nitro-lib#v2.9.0 
```

Reply:
Should be fixed in latest version using parent-level hoisting